.github/workflows/docker-image.yml

```diff
@@ -1,41 +1,55 @@
-name: Docker Image CI
+name: week-1-ci
 
 on: [push]
 
 jobs:
-
-  build:
-
+  build-and-push-app-ml:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ikosovych
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: week-1/
+          file: week-1/Dockerfile
+          push: true
+          tags: ikosovych/week-1:latest    
+          cache-from: type=registry,ref=ikosovych/week-1:buildcache
+          cache-to: type=registry,ref=ikosovych/week-1:buildcache,mode=max
+
+  build-python:
     runs-on: ubuntu-latest
-
     strategy:
       matrix:
-        # Set the Python version(s) to run the tests with
-        python-version: [3.7, 3.8]
+        python-version: [3.7]
```

Comment:
nice nice

btw why 3.7? there is already 3.10

Reply:
I prefer to use older version, because they had less conflicts with versions.